### PR TITLE
[TextFields] "SimpleTextField" Example Prototype

### PR DESCRIPTION
**Pull request by andrewoverton**
This PR adds an experimental `UITextField` subclass called `SimpleTextField` and accompanying examples to the `MaterialComponentExamples` (i.e. not `MaterialComponents` or `MaterialComponentsAlpha`) pod.

The ultimate aim of this text field is to complement (not replace) `MDCTextField` by:
- Providing an alternative when ongoing and difficult to fix `MDCTextField` bugs are blockers for clients/potential clients.
- Providing an alternative for clients who prefer not to work with `MDCTextInputControllers` due to perceived overhead.
- Hopefully doing two things (being a simple or filled text field) very well, rather than trying to do the many things `MDCTextField` and its related classes do.

Things I'm still thinking about/working on:

- VoiceOver. I haven't thought about VoiceOver at all yet. It would be nice to be able to just do what super would do and append the underline labels' and side views' accessibilityLabels/Values to that.
- Should it have some kind of input chip functionality? I'd say this is doable, but would require some thinking.
- Text field density support - The spec mentions "high-density" text fields--this refers to text fields whose bounds fit more snugly around the text, etc. My best idea for achieving this involves a `CGFloat` property called `desiredMainContentHeight`. "Main content height" refers to the area of the text field that contains the text, left/right view, placeholder, clear button, etc. Basically everything but the underline labels. With this property set to 0 (the default) the text field layout would be determined as it is right now. If set to anything other than that, the layout calculation would use the assigned value for the main area height, rather than the value it calculates.
- The manual layout example has some scroll view weirdness that I haven't had time to really fix yet. Also, the Swift storyboard example is basically useless because I can't refer to `SimpleTextField`, an Objective-C class in the same module, from it. I haven't had time to figure this out yet.
- It would be cool if some of the properties supported IBDesignable. I don't know IBDesignable that well, but I think doing this might make it so that you could specify outlined or filled in the storyboard and not have to write any code to achieve either style.
- This isn't code related, but I want to move away from this "Utils" file. It's a catch-all for things that could be more thoughtfully structured. The file structure in general really needs some work.
- There are no tests!

Feel free to look at the examples! "Simple Text Field (Manual Layout)" is better than "Simple Text Field (Storyboard)" right now. 

Here's an LTR video:
simple-text-field-ltr

And an RTL video:
simple-text-field-rtl

### CODE REVIEW

**Comment by yarneo** [comment]
❤️ ❤️ 😍 😸

File: components/schemes/Color/src/MDCSemanticColorScheme.m [hunk 1]
```diff
@@ -67,10 +67,10 @@ + (UIColor *)blendColor:(UIColor *)color withBackgroundColor:(UIColor *)backgrou
   CGFloat bRed = 0.0, bGreen = 0.0, bBlue = 0.0, bAlpha = 0.0;
   [backgroundColor getRed:&bRed green:&bGreen blue:&bBlue alpha:&bAlpha];
 
-  return [UIColor colorWithRed:blendColorChannel(red, bRed, alpha, bAlpha)
+  return [UIColor colorWithRed:blendColorChannel(red, bRed, alpha, bAlpha)  //
```
Comment (rami-a) [inline components/schemes/Color/src/MDCSemanticColorScheme.m]: _Nit:_ Extra `//` at the end

**Review by jverkoey — APPROVED** [review]
This is huge! Thanks for introducing the experimental concept as well - that's really helpful for communicating the state of this code.

Let's get CI passing and then we're go for merge!

**Comment by jverkoey** [comment]
And to clarify, I mean huge as in awesome 🎉

**Comment by andrewoverton** [comment]
Thanks! Sounds good, I might push up one or two small commits and then try to get it passing.

**Pull request merged**